datafusion/core/tests/test_adapter_updated.rs

```diff
@@ -0,0 +1,214 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Due to 
1. THe  previously mentioned reasons that more targets results in longer compile times
2. It will be harder to find tests for the same feature if they are in different files

 I think we should remove this new file `datafusion/core/tests/test_adapter_updated.rs` and put the test in  `datafusion/core/tests/integration_tests/schema_adapter_integration_tests.rs` instead

However, we can do that as a follow on PR too